clang/test/CodeGenSYCL/simple-sycl-virtual-function.cpp

```diff
@@ -0,0 +1,39 @@
+// This test checks that the FE generates global variables corresponding to the
+// virtual table in the global address space (addrspace(1)) when
+// -fsycl-allow-virtual-functions is passed.
+
+// RUN: %clang_cc1 -O0 -triple spir64 -fsycl-allow-virtual-functions -fsycl-is-device -internal-isystem %S/Inputs -emit-llvm %s -o - | FileCheck %s
+
+// CHECK: @_ZTVN10__cxxabiv120__si_class_type_infoE = external addrspace(1) global ptr addrspace(1)
+// CHECK: @_ZTVN10__cxxabiv117__class_type_infoE = external addrspace(1) global ptr addrspace(1)
+// CHECK: @_ZTI4Base = linkonce_odr constant { ptr addrspace(1), ptr } { ptr addrspace(1) getelementptr inbounds (ptr addrspace(1), ptr addrspace(1) @_ZTVN10__cxxabiv117__class_type_infoE, i64 2)
+// CHECK: @_ZTI8Derived1 = linkonce_odr constant { ptr addrspace(1), ptr, ptr } { ptr addrspace(1) getelementptr inbounds (ptr addrspace(1), ptr addrspace(1) @_ZTVN10__cxxabiv120__si_class_type_infoE, i64 2)
+
+#include "sycl.hpp"
+
+SYCL_EXTERNAL int rand();
+
+class Base {
+   public:
+    virtual void display() {}
+};
+
+class Derived1 : public Base {
+   public:
+    void display() {}
+};
+
+int main() {
+  sycl::queue q;
+  q.submit([&](sycl::handler &cgh) {
+    cgh.single_task([=]() {
+      Derived1 d1;
+      Base *b = nullptr;
+      if (rand() % 100 > 50)
+        b = &d1;
+      b->display();
+    });
+  });
+  q.wait();
+  return 0;
+}
```

Comment:
I think we can simplify the test by removing SYCL runtime classes and include sycl.hpp.
```suggestion
SYCL_EXTERNAL void test() {
  Derived1 d1;
  Base *b = nullptr;
  if (rand() % 100 > 50)
    b = &d1;
  b->display();
}
```

Reply:
Done d84e31f